.github/workflows/gpu-ci.yml

```diff
@@ -0,0 +1,33 @@
+# Runs the test suite on a self-hosted GPU machine with CUDA enabled
```

Comment:
Nit: we're only building here, not testing. Is the latter intended for later or not a priority?